Library/src/Multiplayer/SpaceEntity.cpp

```diff
@@ -1339,6 +1339,20 @@ void SpaceEntity::ResolveParentChildRelationship()
 		// Set our new parent
 		Parent = GetSpaceEntitySystem()->FindSpaceEntityById(*ParentId);
 
+		if (Parent == nullptr)
+		{
+			// Parent may not have been added yet
+			// so check pending entities
+			for (const auto& PendingParent : *EntitySystem->PendingAdds)
+			{
+				if (PendingParent->Id == *ParentId)
+				{
+					Parent = PendingParent;
+					break;
+				}
+			}
+		}
```

Comment:
[Consider]
Could `FindSpaceEntityById` be a better place to put this pending check behaviour? It strikes me that this might be the sensible default behaviour?

Perhaps an interface like
`SpaceEntity* SpaceEntitySystem::FindSpaceEntityById(uint64_t EntityId, bool includePending = true);`

Reply:
I believe this will cause a breaking change for clients as I dont think c++ optionals are supported by the wrapper gen

Reply:
I want to address the issue with handling entities when we start planning the multiplayer refactor